modules/i18n/input-method/default.nix

```diff
@@ -0,0 +1,100 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.i18n.inputMethod;
+
+  gtk2Cache = pkgs.runCommandLocal "gtk2-immodule.cache" {
+    buildInputs = [ pkgs.gtk2 cfg.package ];
+  } ''
+    mkdir -p $out/etc/gtk-2.0/
+    GTK_PATH=${cfg.package}/lib/gtk-2.0/ gtk-query-immodules-2.0 > $out/etc/gtk-2.0/immodules.cache
+  '';
+
+  gtk3Cache = pkgs.runCommandLocal "gtk3-immodule.cache" {
+    buildInputs = [ pkgs.gtk3 cfg.package ];
+  } ''
+    mkdir -p $out/etc/gtk-3.0/
+    GTK_PATH=${cfg.package}/lib/gtk-3.0/ gtk-query-immodules-3.0 > $out/etc/gtk-3.0/immodules.cache
+  '';
+
+in {
+  imports =
+    [ ./fcitx.nix ./fcitx5.nix ./hime.nix ./kime.nix ./nabi.nix ./uim.nix ];
+
+  options.i18n = {
+    inputMethod = {
+      enabled = mkOption {
+        type = types.nullOr
+          (types.enum [ "fcitx" "fcitx5" "nabi" "uim" "hime" "kime" ]);
+        default = null;
+        example = "fcitx";
+        description = ''
+          Select the enabled input method. Input methods is a software to input symbols that are not available on standard input devices.
+
+          Input methods are specially used to input Chinese, Japanese and Korean characters.
+
+          Currently the following input methods are available in Home Manager:
+
+          <variablelist>
+          <varlistentry>
+            <term><literal>fcitx</literal></term>
+            <listitem><para>
+              fcitx: A customizable lightweight input method
```

Comment:
Looks better but the `fcitx:`, `fcitx5:`, … seems unnecessary:

![2021-06-20_23-37](https://user-images.githubusercontent.com/798147/122689125-bf960b00-d220-11eb-8fd5-9126bbbfc4e2.png)

Reply:
Why do you think fcitx and fcitx5 are unnecessary?

Reply:
Oh i got it now

Reply:
Yeah im not sure if what you said is necessary or not...

Reply:
Removed.